rust/main/agents/relayer/src/msg/metadata/message_builder.rs

```diff
@@ -206,225 +191,277 @@ mod test {
+        let message_builder = {
+            let mut builder =
+                MessageMetadataBuilder::new(Arc::new(base_builder), ism_address, &message)
+                    .await
+                    .expect("Failed to build MessageMetadataBuilder");
+            builder.max_ism_count = 0;
+            builder
+        };
+
+        let params = MessageMetadataBuildParams::default();
+        let err = build_message_metadata(message_builder, ism_address, &message, params.clone())
             .await
-            .expect("Metadata building failed");
-
-        match res.metadata {
-            Metadata::Found(_) => {
-                panic!("Metadata found when it should have failed");
-            }
-            Metadata::Failed(err) => {
-                assert_eq!(err, MetadataBuildError::MaxIsmDepthExceeded(ISM_MAX_DEPTH));
-            }
-        }
+            .err()
+            .expect("Metadata found when it should have failed");
+        assert_eq!(err, MetadataBuildError::MaxIsmCountReached(0));
+        assert_eq!(*(params.ism_count.lock().await), 0);
     }
 
     #[tokio::test]
-    async fn max_ism_count_reached() {
+    async fn max_depth_reached() {
         let base_builder = build_mock_base_builder();
-
-        for _ in 0..11 {
-            insert_mock_routing_ism(&base_builder);
-        }
+        insert_ism_test_data_001(&base_builder);
 
         let ism_address = H256::zero();
         let message = HyperlaneMessage::default();
 
-        let message_builder =
-            MessageMetadataBuilder::new(Arc::new(Box::new(base_builder)), ism_address, &message)
-                .await
-                .expect("Failed to build MessageMetadataBuilder");
-
-        *message_builder.ism_count.lock().await = ISM_MAX_COUNT - 10;
-
-        let res = build_message_metadata(message_builder, ism_address, &message)
+        let message_builder = {
+            let mut builder =
+                MessageMetadataBuilder::new(Arc::new(base_builder), ism_address, &message)
+                    .await
+                    .expect("Failed to build MessageMetadataBuilder");
+            builder.max_ism_depth = 2;
+            builder
+        };
+
+        let params = MessageMetadataBuildParams::default();
+        let err = build_message_metadata(message_builder, ism_address, &message, params.clone())
             .await
-            .expect("Metadata building failed");
-
-        match res.metadata {
-            Metadata::Found(_) => {
-                panic!("Metadata found when it should have failed");
-            }
-            Metadata::Failed(err) => {
-                assert_eq!(err, MetadataBuildError::MaxIsmCountReached(ISM_MAX_COUNT));
-            }
-        }
+            .err()
+            .expect("Metadata found when it should have failed");
+        assert_eq!(err, MetadataBuildError::AggregationThresholdNotMet(2));
+        assert!(*(params.ism_count.lock().await) <= 4);
```

Comment:
if you add an error log where the depth is checked, you can the make sure the depth error was hit by adding a check like this below (the fn is auto-included if you decorate with `#[tracing_test::traced_test]`)
```rust
        logs_contain("depth limit reached (5)");
```

Reply:
ah cool, I didn't know tracing test provided these :100: